packages/mobile/src/qrcode/utils.ts

```diff
@@ -158,3 +167,30 @@ export function* handleBarcode(
 
   yield call(handleSendPaymentData, qrData, cachedRecipient, isOutgoingPaymentRequest, true)
 }
+
+type PaymentDeepLinkHandlers = {
+  [key in PaymentDeepLinkHandler]: (barcode: QrCode) => Generator
+}
+
+const paymentDeepLinkHandlers: PaymentDeepLinkHandlers = {
+  '': paymentDeepLinkHandlerDefault,
+  merchant: paymentDeepLinkHandlerMerchant,
+}
+
+function* paymentDeepLinkHandlerDefault(barcode: QrCode) {
+  yield put(showError(ErrorMessages.QR_FAILED_INVALID_ADDRESS))
```

Comment:
I don't understand this, if the handler comes from a feature flag why would we put an invalid value there? It looks like the only real option is `merchant`, will there be other options in the future?

In any case, the empty string value we can rename to `disabled` maybe?